### ISSUE TITLE
Staging Review finding: Alternative approach to screen reader user discoverability

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#105281]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** MHV
**Product:** Medications
**Feature:** Filtering in product education (IPE)

## Findings details
**VA.gov Experience Standard - issue:** [User can perceive all meaningful elements.](https://design.va.gov/about/experience-standards/#user-can-perceive-all-meaningful-elements)
**VA.gov Experience Standard - category:** [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

On [/my-health/medications/](https://staging.va.gov/my-health/medications/?page=1), the IPE is associated with the accordion element it's describing using the `aria-describedby` attribute. The goal is to make the IPE more discoverable for screen reader users so they get that information before they interact with the element rather than afterwards. It's currently coded as:

```
<va-accordion ... aria-describedby="rx-ipe-filtering-description">...</va-accordion>

<div id="rx-ipe-filtering-container" data-testid="rx-ipe-filtering-container" ... >...</div>
```

This is something that I wholeheartedly endorsed when you came through previous touchpoints... and I shouldn't have. I wasn't thinking through the full implications of that.

Funny things happen when you add a property to a web component that's not supported by the web component, especially ARIA attributes. Different browsers represent web components differently in the accessibility tree, and different screen readers will interpret the accessibility tree differently. The net result is that for a decent chunk of screen reader users, the `aria-describedby` association will be entirely ignored by their devices.

I don't see a great way around this, but I do have a suggestion for an alternative coding approach that's not a one-to-one match but does boost findability.

## Recommended action

Try something like:
```
<va-accordion ...>...</va-accordion>

<aside id="rx-ipe-filtering-container" aria-label="Hint for filter list" ... >...</aside>
```

You don't get the same direct association between the element being described and the IPE. But placing it in an `aside` will:
- Make it a discoverable landmark
- Give it an `aria-label` (not supported on a `div`) that can describe its purpose
- Ensure that when a screen reader user tabs into the IPE they hear that `aria-label`
- Correctly describe the IPE as complementary content rather than a part of the main content flow

It's not perfect, but behavior should be consistent across screen readers.

I'll also encourage the team to continue iterating on this and design an assistive tech research study to explore this a little more. I think there's a lot of potential with IPEs, but the current iteration shouldn't be the end of the story.

## References

- **Accessibility Defect Severity:** [2: High. Should be fixed in 1-2 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-2)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by TannerBird98** [comment]
Work will be done in this ticket, [https://jira.devops.va.gov/browse/MHV-68722](url)

**Comment by michaelbrodsky-a6** [comment]
Hey @sarahhortonadhoc – thanks so much for your note. It looks like this ticket was cut before our contract began, and this is actually the first I'm seeing it. I just double-checked with @ckeif-iat though and while we don't have production access, we did check staging and it looks like the change described in ticket [#105281](https://github.com/department-of-veterans-affairs/va.gov-team/issues/105281) was added in [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/35837/changes). So he was able to confirm that the behavior described in this ticket (i.e. reading the aria label, describing the IPE as "complementary") is how the page currently works in main today. All this to say, it looks like this is in the codebase and the ticket has officially been addressed.
